src/apps/vpn/appsettingslist.h

```diff
@@ -2,32 +2,8 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-// 1. Define the SETTING macro
-// 2. include this file
-// 3. undefine the SETTING macro
-
-#if defined(_MSVC_TRADITIONAL) && _MSVC_TRADITIONAL
-#  error No supported
-#endif
-
-#define SETTING_BOOL(getter, ...) SETTING(bool, toBool, getter, __VA_ARGS__)
-
-#define SETTING_BYTEARRAY(getter, ...) \
-  SETTING(QByteArray, toByteArray, getter, __VA_ARGS__)
-
-#define SETTING_DATETIME(getter, ...) \
-  SETTING(QDateTime, toDateTime, getter, __VA_ARGS__)
-
-#define SETTING_INT(getter, ...) SETTING(int, toInt, getter, __VA_ARGS__)
-
-#define SETTING_INT64(getter, ...) \
-  SETTING(qint64, toLongLong, getter, __VA_ARGS__)
-
-#define SETTING_STRING(getter, ...) \
-  SETTING(QString, toString, getter, __VA_ARGS__)
-
-#define SETTING_STRINGLIST(getter, ...) \
-  SETTING(QStringList, toStringList, getter, __VA_ARGS__)
+// NOTE! This file is not meant to be used directly. Please use settingslist.h
+// instead.
```

Comment:
Do you mean, don't #include appsettingslist.h but include settingslist.h instead. 

What if you want to add another setting?

I'm probably being over specific but is what you mean:
* you can update this file if needed
* but include settingslist.h

Reply:
Correct. I'll update the comment. We have 2 level of settings:
- shared (settingslist.h)
- per-app (appsettingslist.h)
- per-module (not implemented yet)

The "include" diagram is the following:

1. settingsholder includes both settingslist.h
2. settingslist.h defines the macros, the shared-settings and includes the appsettingslist.h
3. appsettingslist.h defines the per-app settings.